### PR TITLE
Implement a quick find regex for license matching

### PR DESCRIPTION
**Pull request by goneall**
Potential fix for #165

It seems to speed things up - not quite fast enough to get rid of the "slow tests" flags, but a noticeable improvement.

@pmonks - review / give it a try and let me know how it does.  This implements the algorithm we were discussing.

### CODE REVIEW

**Comment by pmonks** [comment]
Reviewing now. Thanks for turning this around so fast @goneall!

**Comment by pmonks** [comment]
Ok so running my default (smaller) test suite here took 593 seconds with v1.1.10, and only 35 seconds with the `issue165` branch.  I also ran it with my full (larger) test suite (which normally takes around an hour with v1.1.10), and with the `issue165` branch it took just over 2 minutes.  All tests were run with pre-downloaded & cached v3.22 SPDX license list and associated assets (so the time to load those assets was negligible - less than a second each time).

**tl;dr: this is a MASSIVE performance improvement!!**  :tada:

**Review by pmonks — APPROVED** [review]

**Comment by goneall** [comment]
@pmonks - Thanks for the review.  Wow!  More of an improvement than I thought.  Thanks for the design suggestions.

I'll go ahead and merge it in.
